### PR TITLE
[Refac] Torch sample as no_grad

### PR DESCRIPTION
**Pull request by dominikandreasseitz**
- Move no_grad call  to backend pyq level
- Remove dead code from engine introduction

### CODE REVIEW

File: qadence/engines/torch/differentiable_expectation.py [hunk 1]
```diff
@@ -232,160 +229,3 @@ def construct_rules(
                 # Since they are constants their gradients are 0.
                 param_to_psr[param_id] = lambda x: torch.tensor([0.0], requires_grad=False)
         return param_to_psr
-
-
-class DifferentiableBackend(Module):
-    """A class to abstract the operations done by the autodiff engine.
-
-    Arguments:
-        backend: An instance of the QuantumBackend type perform execution.
```
Comment (awennersteen) [inline qadence/engines/torch/differentiable_expectation.py:241]: This one is now one level up right, so it was dead code?

**Review by awennersteen — APPROVED** [review]
lgtm

**Comment by awennersteen** [comment]
Thanks!
I'm happy that I hadn't gone totally crazy trying to track the execution path

File: qadence/engines/torch/differentiable_expectation.py (hunk 1 above)
Comment (dominikandreasseitz) [reply qadence/engines/torch/differentiable_expectation.py:241]: we have a "abstract" DifferentiableBackend which is used by both jax and torch. so i guess some remnants werent properly cleaned up

Comment (dominikandreasseitz) [reply qadence/engines/torch/differentiable_expectation.py:241]: resolving so i can merge

**Pull request merged**